README
Crypt/OpenSSL/X509 version 1.9.9
===============================

The README is used to introduce the module and provide instructions on
how to install the module, any machine dependencies it may have (for
example C compilers and installed libraries) and any other information
that should be provided before the module is installed.

A README file is required for CPAN modules since CPAN extracts the
README file from a module distribution so that people browsing the
archive can use it get an idea of the modules uses. It is usually a
good idea to provide version information here so that people can
decide whether fixes for the module are worth downloading.

INSTALLATION

To install this module type the following:

   perl Makefile.PL
   make
   make test
   make install

DEPENDENCIES

This module requires these other modules and libraries:

  OpenSSL Development Libs

COPYRIGHT AND LICENCE

Put the correct copyright and licence information here.

Copyright (C) 2004-2021 Dan Sully

This library is free software; you can redistribute it and/or modify
it under the same terms as Perl itself.


Changes
Revision history for Perl extension Crypt::OpenSSL::X509.

1.9.12

- Applied patch from @jrouzierinverse PR [#93](https://github.com/dsully/perl-crypt-openssl-x509/issues/93)

- Applied patch from @timlegge PR [#92](https://github.com/dsully/perl-crypt-openssl-x509/issues/92)

1.9.11 Mon Dec 13 19:53:39 CET 2021

- Correction to spelling found Debian Linter, thanks @fschlich

- Added eliminated compound-token-split-by-macro errors coming from newer clang/LLVM version (>11?), got some good pointers from this Perl issue https://github.com/Perl/perl5/issues/18780

- Forced OpenSSL under Homebrew to 1.1 via openssl\@1.1, since OpenSSL version 3 got released we might experience issues and this need to be revisited and tested thoroughly

- Reformatted the Changes file, slowly converting to Markdown

1.9.10 Sun Aug  1 09:48:08 CEST 2021

- MANIFEST was not updated with the latest contributions from 1.9.9

  REF: https://github.com/dsully/perl-crypt-openssl-x509/issues/89

1.9.9 Sat Jul 31 17:58:56 CEST 2021

- Contribution by Patrick Cernko. The email method has been extended to return multiple email addresses if available.
  The addresses are concatenated using space (' ') as seperator in order for consumers to extract the multiple email addresses

  REF: https://github.com/dsully/perl-crypt-openssl-x509/pull/88

1.9.8 Thu May 13 18:04:17 CEST 2021

- Addressed minor issue, via PR #87, with the implementation added in 1.9.3 - Thanks Shoichi Kaji

1.9.7 Sun May  2 19:36:24 CEST 2021

- Addressed minor issue with META.yml file not reporting correct version, see issue #86

  REF: https://github.com/dsully/perl-crypt-openssl-x509/issues/86

1.9.6 Sat Apr 24 09:17:00 CEST 2021

- I fell over the CPAN release checklist at https://github.com/Tux/Release-Checklist/blob/master/Checklist.md it mentions
  Devel::PPPort https://metacpan.org/pod/Devel::PPPort. I have now put this to use, raised a single warning

  *** WARNING: Uses is_utf8_string_loclen, which may not be portable below perl 5.9.3, even with 'ppport.h'
  *** Uses 5 C++ style comments, which is not portable
  Analysis completed (1 warning)

  And provided as single patch, which has now been applied and C++ style comments have been changed to C style comments

1.9.5 Thu Apr 22 07:32:16 CEST 2021

- I broke the build for Linux

  - CPAN testers reports: http://matrix.cpantesters.org/?dist=Crypt-OpenSSL-X509+1.904

    The issue is that the change introduced in 1.9.4 introduces an option, which is LLVM specific
    and is not understood by GCC.

    See also issue: #84, ref: https://github.com/dsully/perl-crypt-openssl-x509/issues/84

    I have rearranged the use of flags and try with a match on the GCC version string, which
    can contain the substring LLVM

1.9.4 Wed Apr 21 07:45:58 CEST 2021

- Made a minor change to the Makefile.PL addressing issue with breaking builds on FreeBSD and OpenBSD

  For Perl versions below or equal to 5.20, the error:

  error: nonnull parameter 'pv' will evaluate to 'true' on first encounter [-Werror,-Wpointer-bool-conversion]
  if (pv && len > 1) {

  Has been observed this is now suppressed with converting the error handling into a warning

  See CPAN testers reports:

  - http://www.cpantesters.org/cpan/report/119b4298-9e42-11eb-84bc-edd243e66a77
  - http://www.cpantesters.org/cpan/report/77bdcdd2-a0e7-11eb-84bc-edd243e66a77
  - http://www.cpantesters.org/cpan/report/fd7e66b6-a14b-11eb-84bc-edd243e66a77

1.9.3 Thu Apr  8 07:43:10 CEST 2021

- Addressed issue #81 based on proposed patch from Shoichi Kaji

1.9.2 Thu Nov 12 07:27:56 CET 2020

- Addressed issue #73 via PR #73 removing and excess use of free

1.9.1 Fri Nov  6 19:53:21 CET 2020

- Corrected version number format to address issue #77 via PR #78

1.9 Thu Nov  5 21:38:43 CET 2020

- Bumped Perl minimum requirement from Perl 5.005 to 5.8 PR #76

- Changed from use vars definition to the more modern our PR: #75 Thanks to Todd Rinaldo

- Changed from DynaLoader to XSLoader PR: #75 Thanks to Todd Rinaldo

1.8.13 Thu Oct 24 21:23:46 CEST 2019

- Ensure /usr/local is ahead of /usr in include and lib searches, PR: #74

1.8.12 Thu Nov 22 19:54:37 CET 2018

- Applied patch from @eserte addressing issue (#71) with current directory no longer included in @INC by default from Perl 5.26

  REF: https://www.effectiveperlprogramming.com/2017/01/v5-26-removes-dot-from-inc/

1.8.11 Sun Oct 28 20:22:59 CET 2018

- Re-release of 1.8.10, with corrected version number indexer error from PAUSE

1.8.10 Sun Oct 28 16:52:50 CET 2018

- Maintenance release, corrected issue with MYMETA.* files included in distribution, ref:

  REF: https://weblog.bulknews.net/stop-shipping-mymeta-to-cpan-b92215a227f6

1.8.9 Tue May 30 2017

- Patch / PR from kmx improving detection of OpenSSL libraries under strawberry Perl

1.8.8 Fri Nov 10 2017

- Patch from pi-rho exposing the Issuer's name hash; provide subject_hash() as an alias to hash()

- Patch from stphnlyd X509_get0_signature() was introduced to openssl since 1.0.2.

  REF: https://www.openssl.org/docs/man1.1.0/crypto/X509_get0_signature.html

- Patch from brandond fixing compilation on OpenSSL 1.0.1e

- Patch to support compilation on MacOS Homebrew installed libraries by jonasbn

- Patch from ppisar, patch redefines the accessors only with OpenSSL older than 1.1.0.

- Patch from Sebastian Andrzej Siewior fixing compilation against openssl 1.1.0 and keeping it working against openssl 1.0.2j

- Patch from jonasbn reinitializing inc/ using Module::Install 1.16, fixed issue with META.yml version since META.yml was not regenerated

1.8.7  Thu May 12 2016

- Patch from Bernhard M. Wiedemann to fix compilation errors.

1.8.6  Sat Jan 24 2015

- Patch from James Hunt to print OpenSSL version during tests.
- Various MANIFEST fixes.

1.8.5  Sat Nov 22 2014

- Patch from Uli Scholler to expose more SHA1 hash functions.

1.8.4  Sun Dec 1 2013

- Fix Github Issues #16, #29 & #30
- Possibly fix issue #31

1.8.3  Mon Aug 12 05:50:58 PDT 2013


Makefile.PL
use lib '.';
use inc::Module::Install;
use Module::Install::AutoInstall;

use File::Spec;
use Config;

name('Crypt-OpenSSL-X509');
license('perl');
perl_version('5.008');
all_from('X509.pm');

homepage 'https://github.com/dsully/perl-crypt-openssl-x509';
repository 'https://github.com/dsully/perl-crypt-openssl-x509';
bugtracker 'https://github.com/dsully/perl-crypt-openssl-x509/issues';

requires_external_cc();

if ($^O ne 'MSWin32' and my $prefix = `brew --prefix --installed openssl\@1.1 2>@{[File::Spec->devnull]}`) {
  chomp $prefix;
  inc "-I$prefix/include";
  libs "-L$prefix/lib -lcrypto -lssl";
} else {
  inc '-I/usr/local/opt/openssl/include -I/usr/local/include/openssl -I/usr/include/openssl -I/usr/local/include/ssl -I/usr/local/ssl/include';
  libs '-L/usr/local/opt/openssl/lib -L/usr/local/lib -L/usr/lib -L/usr/local/ssl/lib -lcrypto -lssl';
}

my $cc_option_flags = '-O2 -g -Wall -Werror';

if ($Config::Config{gccversion} =~ /llvm/i) {
  $cc_option_flags .= ' -Wno-deprecated-declarations'; # -Wno-compound-token-split-by-macro

  if ($Config::Config{gccversion} =~ /llvm 13/i) {
      $cc_option_flags .= ' -Wno-compound-token-split-by-macro';
  }

  if ($Config::Config{perl_version} <= 20) {
    $cc_option_flags .= ' -Wno-error=pointer-bool-conversion';
  }
}

if ($Config::Config{myuname} =~ /sunos|solaris/i) {
  # Any SunStudio flags?
} else {
  cc_optimize_flags($cc_option_flags);
}

requires 'Convert::ASN1' => '0.33';

auto_install();
WriteAll;
